### DISCUSSION TITLE
Bar Chart --> Move stacked bar values

### DISCUSSION THREAD

**Discussion by VolkrB** [grafana/grafana#38406] [Dashboards, Panels & Visualizations]
I am creating some stacked bar charts. It would be great to have an option to place values inside of the bars instead of on top when stacking the bars. Please see below:

![image](https://user-images.githubusercontent.com/20803912/130277728-309066c3-b566-4b0f-ac46-925004a205af.png)

I've also discovered that (and maybe this is due to my query) when a value is "0". There is an issue of overlapping values:
![image](https://user-images.githubusercontent.com/20803912/130278237-899290f0-d6b4-4244-8dab-7ed5742db407.png)

So far, I've worked around this via setting "if 0 = null" in my query, but I thought it was worth a mention

**Comment by zuchka** [comment]
hi @VolkrB! 👋  

what version of Grafana are you using? Can you provide a panel json so that we can try and reproduce this overlap issue? Sounds like a bug

**Comment by VolkrB** [comment]
Hi Zuchka,

Thanks for the response! I am currently seeing this is Grafana 8.1 (8.1.1 & 8.1.2).

**I think to reproduce it,  you could use a query that results in a table like this:**
col1| col2| col3
  0    |   3   |   2

**my panel json:**

{
  "id": 50,
  "gridPos": {
    "h": 10,
    "w": 12,
    "x": 0,
    "y": 24
  },
  "type": "barchart",
  "title": "Total Individual Fault Summary - Empty",
  "transformations": [
    {
      "id": "organize",
      "options": {
        "excludeByName": {
          "count": false
        },
        "indexByName": {},
        "renameByName": {}
      }
    }
  ],
  "pluginVersion": "8.1.1",
  "fieldConfig": {
    "defaults": {
      "custom": {
        "lineWidth": 3,
        "fillOpacity": 50,
        "gradientMode": "none",
        "axisPlacement": "right",
        "axisLabel": "",
        "axisSoftMin": 0,
        "hideFrom": {
          "tooltip": false,
          "viz": false,
          "legend": false
        }
      },
      "color": {
        "mode": "palette-classic"
      },
      "thresholds": {
        "mode": "absolute",
        "steps": [
          {
            "color": "green",
            "value": null
          },
          {
            "color": "red",
            "value": 80
          }
        ]
      },
      "mappings": []
    },
    "overrides": [
      {
        "matcher": {
          "id": "byName",
          "options": "bottom"
        },
        "properties": [
          {
            "id": "color",
            "value": {
              "fixedColor": "#3274d9",
              "mode": "fixed"
            }
          }
        ]
      },
      {
        "matcher": {
          "id": "byName",
          "options": "top"
        },
        "properties": [
          {
            "id": "color",
            "value": {
              "fixedColor": "dark-orange",
              "mode": "fixed"
            }
          }
        ]
      },
      {
        "matcher": {
          "id": "byName",
          "options": "side"
        },
        "properties": [
          {
            "id": "color",
            "value": {
              "fixedColor": "#067f90c7",
              "mode": "fixed"
            }
          }
        ]
      }
    ]
  },
  "options": {
    "orientation": "horizontal",
    "showValue": "always",
    "stacking": "normal",
    "groupWidth": 0.7,
    "barWidth": 0.85,
    "tooltip": {
      "mode": "single"
    },
    "legend": {
      "displayMode": "list",
      "placement": "bottom",
      "calcs": []
    },
    "text": {}
  },
  "targets": [
    {
      "format": "table",
      "group": [],
      "metricColumn": "none",
      "rawQuery": true,
      "rawSql": "select name, \r\n (count(feature.contributing) filter (where image.view='Side')) as Side,\r\n (count(feature.contributing) filter (where image.view='Top')) as Top,\r\n (count(feature.contributing) filter (where image.view='Bottom'))as Bottom\r\n from feature\r\n join image on feature.image_id=image.id\r\n and image.stage='Empty'\r\n and feature.contributing=true\r\n and $__timeFilter(create_time)\r\n group by name\r\n order by count(feature.contributing)\r\n desc",
      "refId": "A",
      "select": [
        [
          {
            "params": [
              "confidence"
            ],
            "type": "column"
          }
        ]
      ],
      "table": "feature",
      "timeColumn": "none",
      "timeColumnType": "float8",
      "where": [
        {
          "name": "$__unixEpochFilter",
          "params": [],
          "type": "macro"
        }
      ]
    }
  ],
  "datasource": null
}


![image](https://user-images.githubusercontent.com/20803912/130293442-0d18957a-1abf-4ca2-85bf-6ea3d235d846.png)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
